### DISCUSSION TITLE
Constantly getting 403's from the Storefront API

### DISCUSSION THREAD

**Discussion by RaniSputnik** [Shopify/storefront-api-feedback#30] [Help] [answered]
Hi folks, apologies if this is the wrong place to ask this, but I am constantly getting 403 Forbidden errors from the Storefront API and I can't see what I'm doing wrong.

I have setup a public app and converted it to a sales channel.
I have been through the OAuth flow and added the sales channel to a development store of mine.
I am sending requests to `https://{store}.myshopify.com/api/2021-07/graphql.json` using the following code:

```
const res = await fetch(
  `https://${this.shop}.myshopify.com/api/2021-07/graphql.json`,
  {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "X-Shopify-Storefront-Access-Token": this.accessToken,
    },
    body: JSON.stringify({ query: text }),
  }
);
console.log(res.status);
```
Every time I get a 403.

I thought it might be payment settings on my store, so I fiddled with those but nothing.
I tried removing the Sales channel and re-adding it to the store, still nothing.
I haven't tried setting up a second development store, I'll try that next.

I grabbed a request id from the `x-request-id` header in the response from the API: `36789d7e-66b2-4292-b73e-915e9ee2512f`. I'm wondering if there's any way for me to use that for debugging?

Any help much appreciated! Cheers.

**Comment by callosaurus** [comment, marked as answer]
Hey @RaniSputnik,

Thanks for providing the x-request-id! 
Taking a look at the logs, I can see you're using the access token `shp...9cf`. I assume this is the access token your app received after successful OAuth. Have you used that token to obtain a Storefront API-specific access token? If not, check out this tutorial here: [https://shopify.dev/api/storefront/getting-started](https://shopify.dev/api/storefront/getting-started).

**Reply by RaniSputnik** [reply]
Hey thanks @callosaurus it wasn't obvious to me at first that those were two different things, but makes perfect sense now and I've been able to make calls to the storefront API successfully. Thanks for your help!
